ethereum/contracts/BeefyLightClient.sol

```diff
@@ -140,10 +140,9 @@ contract BeefyLightClient {
 
     // We must ensure at least one block is processed every session,
     // so these constants are checked to enforce a maximum gap between commitments.
-    uint64 public constant NUMBER_OF_BLOCKS_PER_SESSION = 2400;
-    uint64 public constant ERROR_AND_SAFETY_BUFFER = 10;
-    uint64 public constant MAXIMUM_BLOCK_GAP =
-        NUMBER_OF_BLOCKS_PER_SESSION - ERROR_AND_SAFETY_BUFFER;
+    // TODO: verify SAFETY_BUFFER
+    uint64 public numberOfBlocksPerSession;
```

Comment:
should probably use immutable for this: https://solidity-by-example.org/immutable/

(immutable also i believe stores the variable within contract code, rather than in storage, so is way cheaper gas cost to access in future, much like a constant)

Reply:
The session length isn't immutable though. It can and probably will change through a runtime upgrade to Polkadot.